pkg/monitortests/kubeapiserver/staticpodinstall/monitortest.go

```diff
@@ -0,0 +1,155 @@
+package staticpodinstall
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	MonitorName = "staicpod-install-monitor"
+)
+
+func NewStaticPodInstallMonitorTest() *monitorTest {
+	analyzers := []analyzer{
+		&staticPodReadinessProbeEventAnalyzer{
+			windowsByPod: map[podKey][]*podWindow{},
+		},
+		&installerPodPLEGEventAnalyzer{
+			windows: map[podKey]*podWindow{},
+		},
+	}
+	return &monitorTest{analyzers: analyzers}
+}
+
+type monitorTest struct {
+	analyzers []analyzer
+	computed  monitorapi.Intervals
+}
+
+func (mt *monitorTest) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) error {
+	return nil
+}
+
+func (mt *monitorTest) CollectData(ctx context.Context, storageDir string, beginning, end time.Time) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+	return nil, nil, nil
+}
+
+func (mt *monitorTest) ConstructComputedIntervals(ctx context.Context, startingIntervals monitorapi.Intervals, recordedResources monitorapi.ResourcesMap, beginning, end time.Time) (monitorapi.Intervals, error) {
+	mt.computed = mt.construct(startingIntervals)
+
+	return mt.computed, nil
+}
+
+func (mt *monitorTest) EvaluateTestsFromConstructedIntervals(ctx context.Context, finalIntervals monitorapi.Intervals) ([]*junitapi.JUnitTestCase, error) {
+	junitTest := &junitTest{
+		name:     "[sig-apimachinery] installer pods should not run concurrently on two or more nodes",
+		computed: mt.computed,
+	}
+
+	framework.Logf("monitor[%s]: found %d intervals of interest", MonitorName, len(junitTest.computed))
+
+	// the following constraints define pass/fail for this test:
+	// a) if we don't find any constructed/computed interval, then
+	// this test is a noop, so we mark the test as skipped
+	// b) we find constructed/computed intervals, but no occurrences of
+	// concurrent situation, this test is a pass
+	// c) otherwise, there is at least one incident of a
+	// concurrent situation, this test is a flake/fail
```

Comment:
How hard would it be to make the test fail when a logging change causes either parser to observe nothing? If we don't see a single PLEG or probe log for any container (not limited to installer pods), would that be a reliable signal that the logs we're looking for have changed somehow?

Reply:
we want to streamline this for other parsers as well, we will do a follow up PR for this.

Reply:
(I know this is common to all the "log grepping" tests. I'd be happy to see an issue that describes the problem instead of delaying this test, which is useful immediately.)

Reply:
https://issues.redhat.com/browse/OCPBUGS-49301